### PR TITLE
Added a model converter

### PR DESCRIPTION
**Pull request by nvchai**
WIP pull request for adding a model converter from the old waveglow model to the new waveglow model format.

### CODE REVIEW

File: inference.py [hunk 1]
```diff
@@ -28,11 +28,12 @@
 from scipy.io.wavfile import write
 import torch
 from mel2samp import files_to_list, MAX_WAV_VALUE
-
+from convert_model import update_model
 
 def main(mel_files, waveglow_path, sigma, output_dir, sampling_rate, is_fp16):
     mel_files = files_to_list(mel_files)
     waveglow = torch.load(waveglow_path)['model']
+    waveglow = update_model(waveglow)
```
Comment (rafaelvalle) [inline inference.py]: let's keep inference.py clean and have people converting their model manually if necessary.

File: glow_old.py [hunk 2]
```diff
@@ -224,11 +224,10 @@ def infer(self, spect, sigma=1.0):
...
     def remove_weightnorm(model):
-        waveglow = copy.deepcopy(model)
+        waveglow = model
         for WN in waveglow.WN:
             WN.start = torch.nn.utils.remove_weight_norm(WN.start)
             WN.in_layers = remove(WN.in_layers)
             WN.cond_layers = remove(WN.cond_layers)
-            WN.res_layers = remove(WN.res_layers)
-            WN.skip_layers = remove(WN.skip_layers)
+            WN.res_skip_layers = remove(WN.res_skip_layers)
```
Comment (rafaelvalle) [inline glow_old.py]: let's keep res_layers and skip_layers and the deep copy

File: glow.py [hunk 3]
```diff
@@ -293,7 +293,7 @@ def infer(self, spect, sigma=1.0):
 
     @staticmethod
     def remove_weightnorm(model):
-        waveglow = copy.deepcopy(model)
+        waveglow = model
```
Comment (rafaelvalle) [inline glow.py]: let's maintain the deepcopy

File: inference.py (hunk 1 above)
Comment (nvchai) [reply inference.py]: I only had manual conversion, but one of the users requested automatic, so added it.

File: glow_old.py (hunk 2 above)
Comment (nvchai) [reply glow_old.py]: The remove_weightnorm function call is after model conversion. The conversion function removes res_layers and skip_layers and adds res_skip_layers.

File: glow.py (hunk 3 above)
Comment (nvchai) [reply glow.py]: The deepcopy fails if the model has been automatically converted previously. If automatic conversion is to be removed we can reinstate deepcopy

File: inference.py (hunk 1 above)
Comment (rafaelvalle) [reply inference.py]: ok. let's not have it to be automatic and have people convert it manually. this function is not part of inference.

File: glow_old.py (hunk 2 above)
Comment (rafaelvalle) [reply glow_old.py]: we're not converting the model automatically and the pre-trained model loads from glow_old.py which has res_layers and keep_layers.

Comment (nvchai) [reply glow_old.py]: The pre-trained model does not work with inference.py out of the box. It complains res_skip_layers not found if I do not change glow_old.py. It is somehow loading from glow.py and not glow_old.py

Comment (nvchai) [reply glow_old.py]: Also the converted model also fails to load with the same error if I do not change glow_old.py

Comment (nvchai) [reply glow_old.py]: The old model fails to load either way: if I change or do not change glow_old.py

**Comment by rafaelvalle** [comment]
Pushing changes through a separate commit.

**Pull request closed without merging**